src/cpp/src/perf_metrics.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/perf_metrics.hpp"
+#include "openvino/openvino.hpp"
+#include <tuple>
+#include <numeric>
+#include <cmath>
+
+namespace {
+
+std::pair<float, float> calc_mean_and_std(const std::vector<float>& durations) {
+    float mean = std::accumulate(durations.begin(), durations.end(), 0.0f) / durations.size();
+    
+    float sum_square_durations = std::accumulate(durations.begin(), durations.end(), 0.0f,
+        [](const float& acc, const float& duration) -> float {
+            return acc + duration * duration;
+        });
+    float std = std::sqrt(sum_square_durations / durations.size() - mean * mean);
+    return {mean, std};
+}
+
+
+} // namespace
+
+namespace ov {
+namespace genai {
+
+float PerfMetrics::get_duration_ms(std::chrono::steady_clock::duration duration) {
+    return std::chrono::duration_cast<std::chrono::milliseconds>(duration).count();
```

Comment:
Don't cast duration until you really need to use the value (for printing). Return duration itself. That ensures the best accuracy. When you divide a duration, change it's representation to float. For example https://github.com/openvinotoolkit/openvino/blob/ffc135cb1240831411799bdb82ecac352c956f22/samples/cpp/benchmark/throughput_benchmark/main.cpp#L19. But your implementation needs an extra step: when mean is computed keep using source units (most likely nanoseconds, but it's unspecified, you can't rely on that) with float or double representation. Cast the duration to `Ms` or any other suitable unit to use `count()` and print.

Reply:
Done. Now durations are stored in microsecond `chrono::duration<float, std::ratio<1, 1000000>>` for better accuracy. If i store them in ms then tokenization/detokenization times can sometimes be 0, with microseconds.

I convert them only when mean/std are calculated.

Reply:
You shouldn't store duration as any of the specified units. Store whatever `time_point_one - time_point_zero` returns. There's also a cast to int representation in constructors. For example
```C++
    auto start_time = std::chrono::steady_clock::now();
    m_pimpl = std::make_unique<StatefulLLMPipeline>(request, tokenizer, generation_config);
    auto stop_time = std::chrono::steady_clock::now();
    m_pimpl->m_load_time_ms = std::chrono::duration_cast<std::chrono::milliseconds>(stop_time - start_time).count();   
```
Which is bad in two ways: the first is described above, the second is that the used int representation in the cast. That's going to divide by 1000, because the default type is usually nanoseconds.